tests/test.lua

```diff
@@ -212,6 +215,10 @@ local cjson_tests = {
       json.encode, { 10 }, true, { '10' } },
     { "Encode string",
       json.encode, { "hello" }, true, { '"hello"' } },
+    { "Encode empty table as array",
+      json.encode, { empty_array }, true, { '[]' } },
+    { "Encode empty table as table",
+      json.encode, { {} }, true, { '{}' } },
```

Comment:
I suggest adding some more complicates cases here. Your last example can be a good candidate, for example.